server/accounts.go

```diff
@@ -895,24 +895,26 @@ func (a *Account) addClient(c *client) int {
 	if a.clients != nil {
 		a.clients[c] = struct{}{}
 	}
-	added := n != len(a.clients)
-	if added {
-		if c.kind != CLIENT && c.kind != LEAF {
-			a.sysclients++
-		} else if c.kind == LEAF {
-			a.nleafs++
-		}
+	// If we did not add it, we are done
```

Comment:
Changes in accounts.go are simply exiting early if the client was not added/removed, instead of checking for the boolean at every following "if" statements.